src/ir.cpp

```diff
@@ -13616,12 +13623,6 @@ static IrInstruction *ir_get_var_ptr(IrAnalyze *ira, IrInstruction *instruction,
 static IrInstruction *ir_analyze_store_ptr(IrAnalyze *ira, IrInstruction *source_instr,
         IrInstruction *ptr, IrInstruction *uncasted_value)
 {
-    if (ptr->value.type->id != ZigTypeIdPointer) {
```

Comment:
That's great that this is no longer a possible compile error. However the next line accesses `ptr->value.data.x_ptr` - can we have an assert here to protect that? `assert(ptr->value.type->id == ZigTypeIdPointer);`